gosling/api.py

```diff
@@ -195,11 +195,28 @@ def display(self):
 
         display(self)
 
-    def save(self, path: Union[pathlib.Path, str]):
-        spec = self.to_dict()
-        html_str = display.spec_to_html(spec)
+    def save(
+        self,
+        path: Union[pathlib.Path, str],
+        format: Union[Literal["json", "html"], None] = None,
+        **kwargs,
+    ):
+        if format is None:
+            suffix = pathlib.Path(path).suffix.lstrip(".")
+            if suffix in ("json", "html"):
+                format = suffix
+            else:
+                raise ValueError("must specify file format: ['html', 'json']")
+
+        if format == "json":
+            data = self.to_json(**kwargs)
+        if format == "html":
```

Comment:
Shouldn't this be `elif`?

```suggestion
        elif format == "html":
```